test/adapter/rabbitmq_adapter_test.exs

```diff
@@ -4,18 +4,42 @@ defmodule Medusa.Adapter.RabbitMQTest do
   alias Medusa.Broker.Message
   alias Medusa.Adapter.RabbitMQ
 
-  setup do
-    Process.register(self, :self)
-    :ok
+  defp random_string do
+    32 |> :crypto.strong_rand_bytes |> Base.encode64
   end
 
-  describe "RabbitMQ" do
+  defp publish(functions, metadata, opts \\ []) do
```

Comment:
this is bit misleading name. consider changing to `publish_consume` or something that would describe that it does whole thing